### PR TITLE
chore: add and configure typescript + configure eslint

### PR DESCRIPTION
**Pull request by Nebulis**
- added typescript
- configured eslint and fixed issues
- migrated ErrorBoundary.test.tsx

### CODE REVIEW

**Comment by Nebulis** [comment]
integration tests are broken, I need to check =)

**Comment by Nebulis** [comment]
Should be fixed

File: src/reducers/certificate.test.js [hunk 1]
```diff
@@ -9,13 +9,15 @@ import reducer, {
...
     it("should reset the state to initial state", () => {
       const prevState = { foo: "bar" };
-      expect(reducer(prevState, resetCertificateState())).toEqual(initialState);
+      expect(reducer(prevState, resetCertificateState())).toStrictEqual(
+        initialState
+      );
     });
   });
-  describe("CERTIFICATE_TEMPLATE_SELECT_TAB", () => {
+  describe("cERTIFICATE_TEMPLATE_SELECT_TAB", () => {
```
Comment (sumitnitsurat) [inline src/reducers/certificate.test.js:20]: small "c" ? any specific reason ?

Comment (Nebulis) [reply src/reducers/certificate.test.js:20]: image

not that I agree but I dont care at all =)

**Comment by sumitnitsurat** [comment]
Looks good to me.Just sonar is failing.

**Pull request merged**